File: backend/app/api/v1/directory.py
"""Cross-brewery directory endpoints: events and food trucks (spec §8)."""

from __future__ import annotations

from fastapi import APIRouter
from sqlalchemy.ext.asyncio import AsyncSession

from app.api.deps import SessionDep
from app.schemas.directory import BreweryEvent, BreweryFoodTruck
from app.services.current import CurrentDataService

router = APIRouter(tags=["directory"])


@router.get(
    "/events",
    response_model=list[BreweryEvent],
    summary="Current events across all breweries",
)
async def list_events(session: AsyncSession = SessionDep) -> list[BreweryEvent]:
    pairs = await CurrentDataService(session).all_events()
    return [
        BreweryEvent(
            brewery_id=brewery.id,
            brewery_name=brewery.name,
            brewery_slug=brewery.slug,
            brewery_state=brewery.state,
            title=event.title,
            date=event.date,
            description=event.description,
        )
        for brewery, event in pairs
    ]


@router.get(
    "/food-trucks",
    response_model=list[BreweryFoodTruck],
    summary="Current food trucks across all breweries",
)
async def list_food_trucks(
    session: AsyncSession = SessionDep,
) -> list[BreweryFoodTruck]:
    pairs = await CurrentDataService(session).all_food_trucks()
    return [
        BreweryFoodTruck(
            brewery_id=brewery.id,
            brewery_name=brewery.name,
            brewery_slug=brewery.slug,
            brewery_state=brewery.state,
            name=truck.name,
            schedule=truck.schedule,
        )
        for brewery, truck in pairs
    ]

File: frontend/services/directory.ts
/** Cross-brewery events and food-truck listings (spec §8). */

import { apiGet } from "@/lib/api";

interface BreweryRef {
  brewery_id: string;
  brewery_name: string;
  brewery_slug: string;
  brewery_state: string | null;
}

export interface DirectoryEvent extends BreweryRef {
  title: string;
  date: string | null;
  description: string | null;
}

export interface DirectoryFoodTruck extends BreweryRef {
  name: string;
  schedule: string | null;
}

export function fetchDirectoryEvents(): Promise<DirectoryEvent[]> {
  return apiGet<DirectoryEvent[]>("/events");
}

export function fetchDirectoryFoodTrucks(): Promise<DirectoryFoodTruck[]> {
  return apiGet<DirectoryFoodTruck[]>("/food-trucks");
}
